### PR TITLE
feat(pipeline): edge-pipeline provider add handlers for edge, center use

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
edge-pipeline provider add handlers for edge, center use

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |   edge-pipeline provider add handlers for edge, center use           |
| 🇨🇳 中文    |     edge-pipeline provider增加边缘中心侧的使用方法         |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/pipeline/providers/edgepipeline_register/interface.go [hunk 1]
```diff
@@ -40,6 +40,8 @@ type Interface interface {
 
 	ShouldDispatchToEdge(source, clusterName string) bool
 	GetEdgeBundleByClusterName(clusterName string) (*bundle.Bundle, error)
+	OnEdge(func(context.Context))
+	OnCenter(f func(context.Context))
```
Comment (sfwn) [inline modules/pipeline/providers/edgepipeline_register/interface.go:44]: - please unify the format
- add comment for each method
  - can invoke multiple times?
  - sync or async?
  - others

File: modules/pipeline/providers/edgepipeline_register/edge_center_use.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	p.forEdgeUse.handlersOnEdge <- f
+}
+
+func (p *provider) OnCenter(f func(context.Context)) {
+	p.mustStarted()
+	p.forCenterUse.handlersOnCenter <- f
+}
+
+func (p *provider) startEdgeCenterUse(ctx context.Context) {
+	p.Lock()
```
Comment (sfwn) [inline modules/pipeline/providers/edgepipeline_register/edge_center_use.go:51]: no need start at your scenario.

init channel at provider.Init, and listen at provider.Run.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/providers/edgepipeline_register/interface.go (hunk 1 above)
Comment (chengjoey) [reply modules/pipeline/providers/edgepipeline_register/interface.go:44]: adjusted and added

File: modules/pipeline/providers/edgepipeline_register/edge_center_use.go (hunk 2 above)
Comment (chengjoey) [reply modules/pipeline/providers/edgepipeline_register/edge_center_use.go:51]: start at Init， so other providers don't need to call `start`

**Pull request merged**